domainservice/src/main/kotlin/com/example/delivery/kafka/OrderUpdateStatusNotificationProducer.kt

```diff
@@ -2,30 +2,20 @@ package com.example.delivery.kafka
 
 import com.example.commonmodels.order.OrderStatusUpdateNotification
 import com.example.internal.api.KafkaTopic
-import org.apache.kafka.clients.producer.ProducerRecord
 import org.springframework.stereotype.Component
 import reactor.core.publisher.Mono
-import reactor.kafka.sender.KafkaSender
-import reactor.kafka.sender.SenderRecord
 import reactor.kotlin.core.publisher.toMono
+import systems.ajax.kafka.publisher.KafkaPublisher
 
 @Component
 class OrderUpdateStatusNotificationProducer(
-    private val updateStatusNotificationSender: KafkaSender<String, ByteArray>,
+    private val kafkaPublisher: KafkaPublisher,
 ) {
     fun notify(notification: OrderStatusUpdateNotification): Mono<Unit> {
-        return updateStatusNotificationSender.send(
-            createMessage(notification)
+        return kafkaPublisher.publish(
+            KafkaTopic.KafkaOrderStatusUpdateEvents.NOTIFICATIONS,
+            notification.userId,
+            notification.toByteArray()
         ).then(Unit.toMono())
```

Comment:
thenReturn
```suggestion
            notification.toByteArray()
        ).thenReturn(Unit)
```

Reply:
Referred to [docs](https://youtu.be/ooOELrGMn14?si=TDg6r6eXb39oYKfU)